### ISSUE TITLE
[C#] Investigate red crosses in shared projects

### ISSUE THREAD

**Issue by JoshEngebretson** [AtomicGameEngine/AtomicGameEngine#1260]
![issuecapture](https://cloud.githubusercontent.com/assets/376203/20988696/a982a210-bc86-11e6-9c9d-5d0a133ed0a3.PNG)

**Comment by Alan-FGR** [comment]
After an initial investigation, I discovered it has nothing to do with the fact the folders are relative... in the projitems file, there's a <Link></Link> entry in the xml code, apparently that's problematic when it contains actual paths... removing the path make it work fine but obviously cause the files to be scattered on the root folder, however, re-crating the folders in the projectname.Shared folder makes the crosses disappear. It also has nothing to do with using slashes or backslashes.

So, an easy way to fix that is just to re-create the folders even if empty in the actual .Shared folder and then linking the files work properly. I think this is a limitation in Visual Studio, you can't add an existing item without previously having the folders setup in the directory, you can only do that by editing the projitems file manually.

Say you have this:
`	<ItemGroup>
		<Compile Include="..\..\..\Resources\Components\Vehicle.cs">
			<Link>Components\Vehicle.cs</Link>`
This is telling VS to use the path Components\Vehicle.cs for that file, but that path doesn't exist in the Shared folder VS is using.

From what I understand the "Link" entry is used only for displaying purposes. The line above is the actual file path and the line below is where the file is going to be 'linked' in the current hierarchy of the shproj, you can mix local and remote files in the folders there, but you can't have remote-only paths.

In other words, I could be wrong but apparently VS doesn't support that kind of configuration, and an easy way to fix that is just re-creating all the folders in the Shared folder, in that case, the Components folder will get the red x out of the box, but if you right-click the Shared folder and click "Add>New Folder" and create a Components folder, the only effect is removing the x mark.

**Comment by Alan-FGR** [comment]
I think an easier way to look at the problem is that VS uses directories hierarchy to display the external files, but the hierarchy has to exist somewhere, and it not necessarily need to have something to do with the directory hierarchy of the external files you're including in your project, they are different things.
The question is... why VS can't include external/existing directories including its contents. I think that is a limitation.

**Comment by Alan-FGR** [comment]
After reading tons of discussions on SO and MSDN, apparently VS doesn't really have that feature, the best you can do is actually include the files individually, the x mark simply means the sub folder doesn't exist, you can have the same result for example by creating and including a folder, and deleting that folder outside VS, same mark appears indicating that path doesn't exist.

**Comment by Alan-FGR** [comment]
On a related note, since this only affects resharper users, maybe no action is necessary only some documentation about it, since it's quite easy to fix manually, the only problem is knowing /how/ to fix. I don't think resharper should stop working when that happens though, it shouldn't imho.

**Comment by Alan-FGR** [comment]
I'm closing this since it's not a problem in Atomic. The relative path warning issue is not related to this.

**Comment by JoshEngebretson** [comment]
Thanks for the R&D, good findings, still holding out for a solution and will look into it too.

**Comment by Alan-FGR** [comment]
@JoshEngebretson I don't think there's really a solution for the problem other than just automating the folder creation. VS simply doesn't support adding folders that don't exist directly in the project path for that purpose. That 'linking' feature works on a file basis and optionally display the linked files in a hierarchy in the project that is represented by the actual directories, there's no other way to have a properly working hierarchical organization I'm afraid.

**Comment by Alan-FGR** [comment]
After some further investigation, I really don't think there's anything we can do besides automating the directory creation. Did you discover anything @JoshEngebretson?

**Comment by JoshEngebretson** [comment]
Yes, automating the directory creation during solution generation is what I intend to try. I also need to install resharper, thanks

**Comment by JoshEngebretson** [comment]
This is fixed in PR #1275 by ensuring that the folders are generated under the shared items during solution creation
